pkg/k8s/resource/resource.go

```diff
@@ -454,6 +454,10 @@ func (r *resource[T]) Events(ctx context.Context, opts ...EventsOpt) <-chan Even
 				debugInfo))
 		}
 
+		// The set of objects emitted to the subscriber. Used to avoid emitting a delete
```

Comment:
Can't comment there, but I think the `Events` function doc comment should state the semantics around retries, given it's pretty subtle

Reply:
Documented.

Reply:
Documented.